scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -618,7 +618,7 @@ def apply_filters(self, **kwargs):
 
             # If there isn't any overlap between the platform_allow list and the platform_scope
             # we set the scope to the platform_allow list
-            if ts.platform_allow and not platform_filter and not integration:
+            if ts.platform_allow and not platform_filter and not integration and not ts.simulation_only:
```

Comment:
so we will never run this case with twister for real hardware? if it is a simulation_only case?

Reply:
I can probably remove this. The filtering done subsequently should be enough.

Reply:
Removed this, still works as planned. This was leftover from some other attempts at hacking this in.